src/routers/task.js

```diff
@@ -92,4 +94,47 @@ router.delete('/tasks/:id', auth, async(req, res) => {
     }
 })
 
+const upload = multer({
+    limits: {
+        fileSize: 1000000
+    },
+    fileFilter(req, file, callback) {
+        if (!file.originalname.match(/\.(jpg|jpeg|png)$/)) {
+            return callback(new Error('Please upload an image document.'))
+        }
+        callback(undefined, true)
+    }
+})
+
+router.post('/tasks/:id/logo', auth, upload.single('logo'), async(req, res) => {
+    try {
+        const task = await Task.findOne({ _id: req.params.id, createdBy: req.user._id })
```

Comment:
https://mongoosejs.com/docs/tutorials/findoneandupdate.html#how-to-use-findoneandupdate-in-mongoose
You should set the `new` option to `true` to return the document after update was applied.

Reply:
Fixed. (I guess)